Mage/src/main/java/mage/abilities/keyword/MentorAbility.java

```diff
@@ -54,3 +57,37 @@ public boolean apply(ObjectSourcePlayer<Card> input, Game game) {
         return sourcePermanent != null && input.getObject().getPower().getValue() < sourcePermanent.getPower().getValue();
     }
 }
+
+class MentorAbilityEffect extends AddCountersTargetEffect {
+
+    MentorAbilityEffect() {
+        super(CounterType.P1P1.createInstance(), Outcome.BoostCreature);
+    }
+
+    private MentorAbilityEffect(final MentorAbilityEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public MentorAbilityEffect copy() {
+        return new MentorAbilityEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (!super.apply(game, source)) {
+            return false;
```

Comment:
looking at the super method, the return value doesn't actually check that counters were added, just that there were affected targets - probably related to #11718

Reply:
Yeah, original ability `AddCountersTargetEffect` must check `addCounter` result code. Moreover `addCounter` code bugged and can return false on really added counter (if one of the counter were replaced):
![shot_240128_180550](https://github.com/magefree/mage/assets/8344157/76326d19-ef26-450e-8418-2503b98a6616)
Possible fix: `return finalAmount > 0;` at the end instead `returnCode`.